app/routes/product-details.$productSlug/route.tsx

```diff
@@ -27,54 +31,91 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
     return json({ product, canonicalUrl });
 };
 
+interface ProductDetailsLocationState {
+    fromCategory?: {
+        name: string;
+        slug: string;
+    };
+}
+
+export const handle: RouteHandle<typeof loader, ProductDetailsLocationState> = {
+    breadcrumb: (match, location) => {
+        const fromCategory = location.state?.fromCategory;
+
+        const productLink = (
+            <Link to={ROUTES.productDetails.to(match.data.product.slug!)} state={{ fromCategory }}>
+                {match.data.product.name!}
+            </Link>
```

Comment:
What do you think about `ProductLink` component?
We have also some product links on the home page. So user can use this component to sell specific product.